posthog/hogql/database/database.py

```diff
@@ -289,11 +289,18 @@ def create_hogql_database(
     warehouse_tables: dict[str, Table] = {}
     views: dict[str, Table] = {}
 
+    for saved_query in DataWarehouseSavedQuery.objects.filter(team_id=team.pk).exclude(deleted=True):
+        views[saved_query.name] = saved_query.hogql_definition()
+
     for table in (
         DataWarehouseTable.objects.filter(team_id=team.pk)
         .exclude(deleted=True)
         .select_related("credential", "external_data_source")
     ):
+        # Skip adding data warehouse tables that are materialized from views (in this case they have the same names)
+        if table.name in views.keys():
+            continue
```

Comment:
Nit: but you should be able to do `if views.get(table.name):` instead here to avoid enumerating the `views` dict for every table